nemo/collections/asr/parts/mixins/asr_adapter_mixins.py

```diff
@@ -56,6 +56,12 @@ def setup_adapters(self):
         if hasattr(self, 'encoder') and isinstance(self.encoder, AdapterModuleMixin):
             supports_adapters |= True
 
+        if hasattr(self, 'decoder') and isinstance(self.decoder, AdapterModuleMixin):
```

Comment:
How about merging these three in one if block?

Reply:
There are a bunch of such if blocks, not sure if it is a good idea to merge all, just a suggestion to consider.

Reply:
Hmm, seperating them makes it clearer. For example, you could have module adapters which are more specific - pre-decoder or pre-lstm or post-lstm decoder adapters. Covering all cases in one giant list would be problematic. This gives freedom to the dev to catch whatever case they want.